src/validateOptions.js

```diff
@@ -22,17 +22,24 @@ const ajv = new Ajv({
 errors(ajv);
 keywords(ajv, ['instanceof', 'typeof']);
 
+const passed = new WeakSet();
+
 const validateOptions = (schema, options, name) => {
-  if (typeof schema === 'string') {
-    schema = fs.readFileSync(path.resolve(schema), 'utf8');
-    schema = JSON.parse(schema);
-  }
+  if (!passed.has(options)) {
```

Comment:
`schema` need to be stored too.

`WeakMap<Schema, WeakSet<Options>>`

Note that Weak doesn't allow string as key. Best skip cache for string values. We can't be sure that the same string results in the same Schema. One could change the file between calls.

I would recommend to remove file reading in the next major version.

Reply:
agreed, since its usage is loader author facing

Reply:
@sokra hmm, i just pushed a similar solution for this - i think this covers when schema is string.

Reply:
@sokra I've gone ahead and applied these changes - which has upgraded this pr to a breaking change. My thinking is this could go in next major release.

Reply:
hmm... shouldn't just marking the options object passed be enough?